### PR TITLE
refactor(components, protocol-designer): update Toolbox implementations

### PR DESCRIPTION
**Pull request by ncdiehl11**
# Overview

Refactors Toolbox and updates its implementations across PD app to more closely mirror designs. Importantly, sets DeckSetupTools implementation to absolute to accommodate full-screen zoom on selected slot/cutout.

## Test Plan and Hands on Testing

Navigate to various implementations of Toolbox and verify that layout matches designs. Specifically:
- DeckSetupTools, on- and off-deck (check out zoom when adding hardware on deck)
- LiquidToolbox
- TimelineToolbox
- SubstepsToolbox
- StepformToolbox

Closes RQA-3262

## Changelog

- update props for Toolbox
- move logic for position styles for absolutely-positioned Toolbox to caller
- restyle all implementations of Toolbox to maintain layout where necessary

## Review requests

- see test plan

## Risk assessment

medium. touches Toolbox component used across PD app

### CODE REVIEW

File: protocol-designer/src/organisms/AssignLiquidsModal/LiquidToolbox.tsx [hunk 1]
```diff
@@ -239,7 +239,7 @@ export function LiquidToolbox(props: LiquidToolboxProps): JSX.Element {
           onClose()
         }}
         onCloseClick={handleClearSelectedWells}
-        height="calc(100vh - 64px)"
+        height="100%"
```
Comment (koji) [inline protocol-designer/src/organisms/AssignLiquidsModal/LiquidToolbox.tsx:242]: 👍

**Review by koji — APPROVED** [review]
lgtm, thank you for refactoring!

File: components/src/organisms/Toolbox/index.tsx [hunk 2]
```diff
@@ -5,31 +5,27 @@ import {
...
+  POSITION_RELATIVE,
 } from '../../styles'
 import { BORDERS, COLORS } from '../../helix-design-system'
 import { SPACING } from '../../ui-style-constants'
 import { PrimaryButton } from '../../atoms'
 import { textDecorationUnderline } from '../../ui-style-constants/typography'
+import type { StyleProps } from '../../primitives'
 
-export interface ToolboxProps {
+export interface ToolboxProps extends StyleProps {
```
Comment (jerader) [inline components/src/organisms/Toolbox/index.tsx:16]: nice this cleans up Toolbox a lot!

File: protocol-designer/src/pages/Designer/DeckSetup/DeckSetupContainer.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { useMemo, useState, Fragment } from 'react'
 import { useDispatch, useSelector } from 'react-redux'
+import { round } from 'lodash'
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/DeckSetup/DeckSetupContainer.tsx]: nit 
```suggestion
import round  from 'lodash/round'
```

File: protocol-designer/src/pages/Designer/DeckSetup/DeckSetupContainer.tsx [hunk 4]
```diff
@@ -122,19 +123,37 @@ export function DeckSetupContainer(props: DeckSetupTabType): JSX.Element {
...
-  const viewBoxWidth = deckDef.dimensions[0]
+  const viewBoxWidth = deckDef.dimensions[0] / deckMapRatio
   const viewBoxHeight = deckDef.dimensions[1] + DETAILS_HOVER_SPACE
 
   const initialViewBox = `${viewBoxX} ${viewBoxY} ${viewBoxWidth} ${viewBoxHeight}`
 
   const [viewBox, setViewBox] = useState<string>(initialViewBox)
 
+  const isZoomed = Object.values(zoomIn).some(val => val != null)
+  const viewBoxNumerical = viewBox?.split(' ').map(val => Number(val)) ?? []
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/DeckSetup/DeckSetupContainer.tsx:146]: do you mind confirming that the zoomed in viewbox looks ok with Mel? Looking at the latest candidate vs your branch, looks like the padding is missing when you zoom in

your branch
Screenshot 2024-12-06 at 14 14 45

vs latest candidate
Screenshot 2024-12-06 at 14 15 05

**Review by jerader — COMMENTED** [review]
looks good! left a few comments, mainly I'd like to confirm that the zoom-in deck map looks good with Mel.

File: protocol-designer/src/pages/Designer/DeckSetup/DeckSetupContainer.tsx (hunk 4 above)
Comment (ncdiehl11) [reply protocol-designer/src/pages/Designer/DeckSetup/DeckSetupContainer.tsx:146]: Screenshot 2024-12-06 at 2 20 59 PM

looks like no padding but I can check with Mel

Comment (jerader) [reply protocol-designer/src/pages/Designer/DeckSetup/DeckSetupContainer.tsx:146]: if you don't mind, that'd be great. thank you! i looked at some designs a few weeks ago to fix the deck map zoomed out and i thought that there was padding when zoomed in too.

Comment (ncdiehl11) [reply protocol-designer/src/pages/Designer/DeckSetup/DeckSetupContainer.tsx:146]: confirmed, good to go

**Pull request merged**